lib/GenomeFileUtil/core/GenbankToGenome.py

```diff
@@ -177,50 +163,391 @@ def stage_input(self, params):
                                 })['node_file_name']
             genbank_file_path = os.path.join(input_directory, file_name)
 
         if 'ftp_url' in file and file['ftp_url'] is not None:
-            # Note that the Transform originally had a script_utils.download_from_urls method
-            # that, if the url is a folder, pulls all subfiles.  That code recently broke when
-            # fetching from NCBI (not clear if it is our issue or NCBI), but for now just
-            # support the most common case- an FTP to a single file.
             print('Downloading file from: ' + str(file['ftp_url']))
-            sys.stdout.flush()
-
-            url = urlparse(file['ftp_url'])
-            if url.scheme != 'ftp' and url.scheme != 'http':
-                raise ValueError('Only FTP/HTTP servers are supported')
-            file_name = 'genome.gbk'
-            if url.path != '':
-                file_name = url.path.split('/')[-1]
-
-            req = urllib2.Request(file['ftp_url'])
-            response = urllib2.urlopen(req)
-            file_data = response.read()
-
-            genbank_file_path = os.path.join(input_directory, file_name)
-            with open(genbank_file_path, "w") as genbank_file:
-                genbank_file.write(file_data)
+            local_file_path = self.dfu.download_web_file({
+                'file_url': file['ftp_url'],
+                'download_type': 'FTP'
+            })['copy_file_path']
+            genbank_file_path = os.path.join(input_directory,
+                                             os.path.basename(local_file_path))
+            shutil.copy2(local_file_path, genbank_file_path)
 
         # extract the file if it is compressed
         if genbank_file_path is not None:
             print("staged input file =" + genbank_file_path)
-            sys.stdout.flush()
-            dfUtil = DataFileUtil(self.cfg.callbackURL)
-            dfUtil.unpack_file({ 'file_path': genbank_file_path })
+            self.dfu.unpack_file({'file_path': genbank_file_path})
 
         else:
             raise ValueError('No valid files could be extracted based on the input')
 
         return input_directory
 
-
-
-
+    def parse_genbank(self, file_path, params):
+        print("Saving original file to shock")
+        shock_res = self.dfu.file_to_shock({
+            'file_path': file_path,
+            'make_handle': 1,
+            'pack': 'gzip',
+        })
+        # Write and save assembly file
+        contigs = Bio.SeqIO.parse(file_path, "genbank")
+        assembly_ref = self._save_assembly(contigs, params)
+        assembly_data = self.dfu.get_objects(
+            {'object_refs': [assembly_ref],
+             'ignore_errors': 0})['data'][0]['data']
+        genome = {
+            "id": params['genome_name'],
+            "type": params['type'],
+            "original_source_file_name": os.path.basename(file_path),
+            "genetic_code": params['genetic_code'],
```

Comment:
If this is not passed in you need to get the Genetic code from the Taxonomy object.